src/main/java/edu/wpi/grip/core/Connection.java

```diff
@@ -79,4 +104,13 @@ public void removeConnection(SourceRemovedEvent e) {
             }
         }
     }
+
+    /**
+     * Thrown when the user tries to create a connection that will result in an infinite loop.
+     */
+    public static class InfiniteLoopException extends Error {
```

Comment:
I beg to differ. Look at the java `URL` object. It throws an exception if the URL is malformed.